src/Components/TodoContainer.js

```diff
@@ -1,74 +1,106 @@
-  const [isLoading, setIsLoading] = useState(true);
-
-  // New async function to fetch data from the API
-  async function fetchData() {
-    // Define options for the API request
-    const options = {
-      method: "GET",
-      headers: {
-        Authorization: `Bearer ${process.env.REACT_APP_AIRTABLE_API_KEY}`,
-      },
-    };
-
-    // Construct the URL for the API
-    const url = `https://api.airtable.com/v0/${process.env.REACT_APP_AIRTABLE_BASE_ID}/${process.env.REACT_APP_TABLE_NAME}`;
-
-    try {
-      const response = await fetch(url, options);
-
-      if (!response.ok) {
-        console.log("Airtable error response:");
-        console.log(response);
-        throw new Error(`Error: ${response.status}`);
-      }
-
-      // Parse the response JSON
-      const data = await response.json();
-
-      // Transform API data.records into todo objects
-      const todos = data.records.map((record) => ({
-        id: record.id,
-        title: record.fields.title,
-      }));
-
-      // Update todoList and isLoading states
-      setTodoList(todos);
-      setIsLoading(false);
-    } catch (error) {
-      console.log(`Fetch error: ${error.message}`);
+    const [todoList, setTodoList] = useState([{ id: '', title: '', createdTime: '' }]);
+    const [isLoading, setIsLoading] = useState(true);
+    const [isAscending, setIsAscending] = useState(true); // Track sorting order
+
+    // Memoize the fetchData function using useCallback to ensure a stable reference
+    const fetchData = useCallback(async () => {
+        // Define options for the API request
+        const options = {
+            method: "GET",
+            headers: {
+                Authorization: `Bearer ${process.env.REACT_APP_AIRTABLE_API_KEY}`,
+            },
+        };
+
+        // Construct the URL for the API with ascending or descending order
+        // Update the URL with sort parameters (field and direction)
+        const order = isAscending ? "asc" : "desc";
+        const sortField = "created"; // Change this to the desired sorting field
+        const url = `https://api.airtable.com/v0/${process.env.REACT_APP_AIRTABLE_BASE_ID}/${process.env.REACT_APP_TABLE_NAME}?sort[0][field]=${sortField}&sort[0][direction]=${order}`;
+
+        try {
+            const response = await fetch(url, options);
+
+            if (!response.ok) {
+                console.log("Airtable error response:");
+                console.log(response);
+                throw new Error(`Error: ${response.status}`);
+            }
+
+            // Parse the response JSON
+            const data = await response.json();
+            console.log('API Response Data:', data);
+
+            // Transform API data.records into todo objects
+            const todos = data.records.map((record) => ({
+                id: record.id,
+                title: record.fields.title,
+                createdTime:new Date (record.fields.createdTime),
```

Comment:
Woops, looks like I used the wrong structure here, and neither of us noticed - sorry about that!

<img width="741" alt="Screen Shot 2023-10-02 at 5 35 12 PM" src="https://github.com/Victoria240/react-todo/assets/12771651/5d5bae49-1670-4d6f-b37e-f3ffed7f9ef7">

It turns out that `createdTime` isn't actually on the `fields` object, but at the top level of the `record`

Reply:
Oh, I just noticed that too. I got confused by the `createdTime` and `created` values 😄. Do both values have to be present? Or do I need to rename the `created` field to `createdTime` ?

> Woops, looks like I used the wrong structure here, and neither of us noticed - sorry about that!
> 
> <img alt="Screen Shot 2023-10-02 at 5 35 12 PM" width="741" src="https://user-images.githubusercontent.com/12771651/272110490-5d5bae49-1670-4d6f-b37e-f3ffed7f9ef7.png">
> It turns out that `createdTime` isn't actually on the `fields` object, but at the top level of the `record`
<img width="283" alt="Screenshot 2023-10-03 at 09 53 36" src="https://github.com/Victoria240/react-todo/assets/66220547/7e84a6b7-fc73-4eea-b204-70980af751a9">